cli/vm/cli.go

```diff
@@ -1045,27 +1085,35 @@ func handleRun(c *cli.Context) error {
 			}
 		}
 		if runCurrent {
-			if m == nil {
+			if cs == nil {
 				return fmt.Errorf("manifest is not loaded; either use 'run' command to run loaded script from the start or use 'loadgo' and 'loadnef' commands to provide manifest")
 			}
-			md := m.ABI.GetMethod(args[0], len(params))
+			md := cs.Manifest.ABI.GetMethod(args[0], len(params))
 			if md == nil {
 				return fmt.Errorf("%w: method not found", ErrInvalidParameter)
 			}
+			hasRet = md.ReturnType != smartcontract.VoidType
 			offset = md.Offset
+			var initOff = -1
+			if initMD := cs.Manifest.ABI.GetMethod(manifest.MethodInit, 0); initMD != nil {
+				initOff = initMD.Offset
+			}
+
+			// QUESTION: @roman-khimov, v.LoadNEFMethod creates new context and pushes it onto invocation stack.
+			// To handle it properly, we need to unload previously loaded context (and it is done). But then
+			// we have to manually reset the gas limit to the previous value and (which is more important)
+			// preserve the same set of breakpoints, see TestRunWithBreakpoint.
+			// Are you sure that we really want to use v.LoadNEFMethod here and change the executable context?
+
+			ic := getInteropContextFromContext(c.App)
+			gasLimit := v.GasLimit
+			ic.ReuseVM(v)         // Clear context loaded by 'loadgo', 'loadnef' or 'loaddeployed' to properly handle LoadNEFMethod.
+			v.GasLimit = gasLimit // Preserve the gas limit set by loading commands.
+			v.LoadNEFMethod(&cs.NEF, util.Uint160{}, cs.Hash, callflag.All, hasRet, offset, initOff, nil)
```

Comment:
@roman-khimov, could you take a look, please?

Reply:
What are the alternatives? LoadNEFMethod is just a (known, correct) way to set things up for a contract. IIRC there was some problem with passing NEF itself into the context for it to work correctly (for example `CALLT`).

Reply:
Fixed.